modelarmor/api/ModelArmor.Samples.Tests/UpdateTemplateTest.cs

```diff
@@ -0,0 +1,102 @@
+    public class UpdateTemplateTests : IClassFixture<ModelArmorFixture>
+    {
+        private readonly ModelArmorFixture _fixture;
+        private readonly CreateTemplateSample _create_template_sample;
+        private readonly UpdateTemplateSample _update_template_sample;
+
+        public UpdateTemplateTests(ModelArmorFixture fixture, ITestOutputHelper output)
+        {
+            _fixture = fixture;
+            _create_template_sample = new CreateTemplateSample();
+            _update_template_sample = new UpdateTemplateSample();
+        }
+
+        [Fact]
+        public void UpdateTemplateTest()
+        {
+            // Create a template.
+            TemplateName templateName = _fixture.CreateTemplateName();
+            _fixture.RegisterTemplateForCleanup(templateName);
+
+            Template originalTemplate = _create_template_sample.CreateTemplate(
+                projectId: _fixture.ProjectId,
+                locationId: _fixture.LocationId,
+                templateId: templateName.TemplateId
+            );
+
+            string templateId = TemplateName.Parse(originalTemplate.Name).TemplateId;
+
+            // Call the sample to modify the RAI filter of above created template.
+            Template updatedTemplate = _update_template_sample.UpdateTemplate(
+                projectId: _fixture.ProjectId,
+                locationId: _fixture.LocationId,
+                templateId: templateId
+            );
+
+            // Assertions.
+            Assert.NotNull(updatedTemplate);
+            Assert.Equal(originalTemplate.Name, updatedTemplate.Name);
+
+            // Verify specific RAI filters were updated.
+            Assert.NotNull(updatedTemplate.FilterConfig);
+            Assert.NotNull(updatedTemplate.FilterConfig.RaiSettings);
+
+            var raiFilters = updatedTemplate.FilterConfig.RaiSettings.RaiFilters;
+            Assert.Contains(raiFilters, f => f.FilterType == RaiFilterType.Dangerous);
+            Assert.Contains(
+                raiFilters,
+                f =>
+                    f.FilterType == RaiFilterType.Dangerous
+                    && f.ConfidenceLevel == DetectionConfidenceLevel.High
+            );
+
+            Assert.Contains(raiFilters, f => f.FilterType == RaiFilterType.HateSpeech);
+            Assert.Contains(
+                raiFilters,
+                f =>
+                    f.FilterType == RaiFilterType.HateSpeech
+                    && f.ConfidenceLevel == DetectionConfidenceLevel.MediumAndAbove
+            );
+
+            Assert.Contains(raiFilters, f => f.FilterType == RaiFilterType.Harassment);
+            Assert.Contains(
+                raiFilters,
+                f =>
+                    f.FilterType == RaiFilterType.Harassment
+                    && f.ConfidenceLevel == DetectionConfidenceLevel.MediumAndAbove
+            );
+
+            Assert.Contains(raiFilters, f => f.FilterType == RaiFilterType.SexuallyExplicit);
+            Assert.Contains(
```

Comment:
These are redundant. If it contains the more specific, then it also contains the least specific. Here and in all the asserts above, keep the most specific only.